pkg/infoschema/infoschema_v2.go

```diff
@@ -331,6 +454,44 @@ func (isd *Data) GCOldVersion(schemaVersion int64) (int, int64) {
 			zap.Bool("byID success", succ1),
 			zap.Bool("byName success", succ2))
 	}
+	isd.gcOldFKVersion(schemaVersion)
+	return len(deletes), total
+}
+
+// GCOldFKVersion compacts btree nodes by removing items older than schema version.
+func (isd *Data) gcOldFKVersion(schemaVersion int64) (int, int64) {
```

Comment:
If it's the same {dbName, tableName, schemaVersion} => fkName[] pattern, we could have extract a same function for that.